align_system/algorithms/outlines_regression_adm.py

```diff
@@ -99,33 +159,61 @@ def sample_outcome_predictions(self,
 
         return predicted_outcomes
 
-    def _format_single_incontext_prompt(self, prompt, target_kdma, expected_value):
+    def _format_single_incontext_prompt(self, icl_example, target_kdma):
         # Predicted scores are 0-10, KDMA values are 0-1
-        expected_value = int(expected_value * 10)
-
-        # TODO: Add chain of thought reasoning?
-        target_wording_mapping = {
-            "MoralDesert": "moral deservingness",
-            "maximization": "maximization"
-        }
-        reasoning_text = f"Per the principle of {target_wording_mapping.get(target_kdma, target_kdma)}, " \
-                         f"this response scores a {expected_value}"
+        expected_value = int(icl_example['expected_value'] * 10)
 
-        # TODO: Validate against schema?
-        answer = f'{{"reasoning": "{reasoning_text}", "score": {expected_value}}}'
+        answer = f'{{"reasoning": "{icl_example['reasoning']}", "score": {expected_value}}}'
 
         return [
-            {"role": "user", "content": prompt},
+            {"role": "user", "content": icl_example['prompt']},
             {"role": "assistant", "content": answer}
         ]
 
+    def get_icl_datasets(self, incontext_settings, target_kdmas):
+        icl_datasets = {}
+        # Read dataset(s)
+        for target_kdma in target_kdmas:
+            dset_kdma = target_kdma['kdma']
+            dset_f = incontext_settings["datasets"][dset_kdma]
+            with open(dset_f) as f:
+                dset = json.load(f)
+
+            icl_datasets[dset_kdma] = []
+            for icl_sample in dset:
+                state, actions = hydrate_scenario_state(icl_sample["input"])
+                icl_choices = self.format_choices(
+                    [a.unstructured for a in actions],
+                    actions,
+                    state
+                )
+                for action, icl_choice, label in zip(actions, icl_choices, icl_sample["label"]):
+                    if dset_kdma not in label:
+                        continue
+
+                    icl_scenario_description = scenario_state_description_dre(state)
+
+                    # TODO: Include outcome in ICL example?
+                    icl_prompt = kdma_score_prediction_prompt(
+                        icl_scenario_description, icl_choices, icl_choice, None, dset_kdma
+                    )
+
+                    icl_datasets[dset_kdma].append({
+                        "prompt": icl_prompt,
+                        "expected_value": label[dset_kdma],
+                        "reasoning": get_chain_of_thought_reasoning(target_kdma, action, state, icl_choice, label[dset_kdma])
```

Comment:
Nit: Every time we use the label value, we multiply it by 10, maybe we should just do it once here?

Reply:
Not worth holding up the PR, though. Merging!